opal/mca/btl/ofi/btl_ofi_component.c

```diff
@@ -0,0 +1,538 @@
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2014-2018 Los Alamos National Security, LLC. All rights
+ *                         reserved.
+ * Copyright (c) 2018      Intel, Inc, All rights reserved
+ *
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+
+#include "opal_config.h"
+
+#include "opal/mca/btl/btl.h"
+#include "opal/mca/btl/base/base.h"
+#include "opal/mca/hwloc/base/base.h"
+
+#include <string.h>
+
+#include "btl_ofi.h"
+#include "btl_ofi_providers.h"
+#include "btl_ofi_rdma.h"
+
+
+static char *prov_include;
+static char *prov_exclude;
+static char *ofi_progress_mode;
+static int mca_btl_ofi_init_device(struct fi_info *info);
+
+static int mca_btl_ofi_component_register(void)
+{
+    mca_btl_ofi_module_t *module = &mca_btl_ofi_module_template;
+
+    /* fi_getinfo with prov_name == NULL means ALL provider.
+     * Since now we are using the first valid info returned, I'm not sure
+     * if we need to provide the support for comma limited provider list. */
+    prov_include = NULL;
+    (void) mca_base_component_var_register(&mca_btl_ofi_component.super.btl_version,
+                                          "provider_include",
+                                          "OFI provider that are considered for use "
+                                          "(e.g., \"psm,psm2\"; an empty value means that all providers will "
+                                          "be considered.",
+                                          MCA_BASE_VAR_TYPE_STRING, NULL, 0, 0,
+                                          OPAL_INFO_LVL_1,
+                                          MCA_BASE_VAR_SCOPE_READONLY,
+                                          &prov_include);
+
+    /* TODO: this param has not been implemented. Not sure if we need it. " */
+    prov_exclude = NULL;
+    (void) mca_base_component_var_register(&mca_btl_ofi_component.super.btl_version,
+                                          "provider_exclude",
+                                          "Comma-delimited list of OFI providers that are not considered for use "
+                                          "(default: \"sockets,mxm\"; empty value means that all providers will "
+                                          " be considered). "
+                                          "Mutually exclusive with btl_ofi_provider_include.",
+                                          MCA_BASE_VAR_TYPE_STRING, NULL, 0, 0,
+                                          OPAL_INFO_LVL_1,
+                                          MCA_BASE_VAR_SCOPE_READONLY,
+                                          &prov_exclude);
+
+    /* Note: better leave it at 1 for now. osc rdma module is designed for 1 completion
+     * at a time. Dealing with more than 1 completion in 1 read will confuse the osc rdma.
+     * source: 8 hours of debugging. :(*/
```

Comment:
Did you talk to @hjelmn about this?

Reply:
I talked to @hppritcha who talked to @hjelmn and they agreed that this is a bug. I haven't file the issue upstream.